controller/WebController.php

```diff
@@ -559,22 +559,34 @@ public function invokeGenericErrorPage($request, $message = null)
      */
     public function invokeChangeList($request, $prop='dc:created')
     {
-        // set language parameters for gettext
-        $this->setLanguageProperties($request->getLang());
-        $vocab = $request->getVocab();
-        $offset = ($request->getQueryParam('offset') && is_numeric($request->getQueryParam('offset')) && $request->getQueryParam('offset') >= 0) ? $request->getQueryParam('offset') : 0;
-        $changeList = $vocab->getChangeList($prop, $request->getContentLang(), $request->getLang(), $offset);
+        $bydate = $this->formatChangeList($request, $prop);
+
         // load template
         $template = $this->twig->loadTemplate('changes.twig');
 
         // render template
         echo $template->render(
             array(
-                'vocab' => $vocab,
+                'vocab' => $request->getVocab(),
                 'languages' => $this->languages,
                 'request' => $request,
-                'changeList' => $changeList)
+                'changeList' => $bydate)
             );
     }
 
+    public function formatChangeList($request, $prop='dc:created') {
```

Comment:
This should be a private function don't you think?

Also here I would not use a default value for `$prop` since it could be either `dc:created` or `dc:modified`, with roughly equal probability

Reply:
Also, the name of the method seems to imply that the function only performs formatting, but it actually also fetches the change list. Either change it to `getFormattedChangeList`, or - perhaps even better - move the call to `$vocab->getChangeList` outside the method, so that it only performs formatting by date.

Reply:
Good point, I will split the method in two. I left the function as public so that it could be included in the unit tests. Would there be a better way?